source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/profile/picture/index.tsx

```diff
@@ -6,46 +6,65 @@ export interface IProps {
 
 export interface IState {
   imageData: string;
+  width: number;
+  height: number;
 }
 
 class ImageProfile extends React.Component<IProps, IState> {
 
   state = {
-    imageData: null
+    imageData: null,
+    width: null,
+    height: null
   };
 
   handleNewImage = e => {
 
     if (e.target.files && e.target.files[0]) {
-      const reader = new FileReader();
 
-      reader.readAsDataURL(e.target.files[0]);
+      const i = new Image();
+      const file = e.target.files[0];
+
+      i.onload = () => {
+        const reader = new FileReader();
+        reader.readAsDataURL(file);
+        reader.onload = () => {
+          this.setState({
+            imageData: reader.result,
+            width: i.width,
+            height: i.height
+          }, () => {
+            this.props.onChange(reader.result);
+          });
+        };
+      }
+
+      i.src = URL.createObjectURL(file);
 
-      reader.onload = () => {
-        this.setState({ imageData: reader.result },() => {
-          this.props.onChange(reader.result);
-        });
-      };
     }
 
 
   }
 
   render() {
 
-    const { imageData } = this.state;
+    const { imageData, width, height } = this.state;
+
+    console.log(width, height);
 
     return (
       <div data-test-picture-profile>
         {!imageData ?
           <div className='default-profile-image'>
             <span>JL</span>
           </div> :
-          <img
-            className='profile-image'
-            src={imageData}
-            data-test-picture-uploaded
-          />
+          <div className='thumbnail'>
```

Comment:
this could probably be extracted to its own component (the thumbnail and the initials (when there is no image)), and we could throw it in the header as well (if we can adjust the size).
we could then wrap it in a `withCurrentUser` and auto-populate the initials (if they exist (maybe defaut to `FL` for first/last? or `??` to encourage people to set it to something? idk)